providers/openfeature-provider-flagd/src/openfeature/contrib/provider/flagd/resolvers/process/connector/grpc_watcher.py

```diff
@@ -0,0 +1,132 @@
+        self.emit_provider_error = emit_provider_error
+
+        self.connected = False
+
+    def create_stub(
+        self,
+    ) -> typing.Tuple[sync_pb2_grpc.FlagSyncServiceStub, grpc.Channel]:
+        config = self.config
+        channel_factory = grpc.secure_channel if config.tls else grpc.insecure_channel
+        channel = channel_factory(
+            f"{config.host}:{config.port}",
+            options=(
+                ("grpc.max_reconnect_backoff_ms", 1000),
+                ("grpc.initial_reconnect_backoff_ms", 1000),
+                ("grpc.keepalive_time_ms", 1000),
+            ),
+        )
+        stub = sync_pb2_grpc.FlagSyncServiceStub(channel)
+        return stub, channel
+
+    def initialize(self, context: EvaluationContext) -> None:
+        self.active = True
+        self.thread = threading.Thread(
+            target=self.sync_flags, daemon=True, name="FlagdGrpcSyncWorkerThread"
+        )
+        self.thread.start()
+
+        ## block until ready or deadline reached
+        deadline = self.timeout + time.time()
+        while not self.connected and time.time() < deadline:
+            time.sleep(0.05)
+        logger.debug("Finished blocking gRPC state initialization")
+
+        if not self.connected:
+            raise ProviderNotReadyError(
+                "Blocking init finished before data synced. Consider increasing startup deadline to avoid inconsistent evaluations."
+            )
+
+    def shutdown(self) -> None:
+        self.active = False
+        self.channel.close()
+
+    def sync_flags(self) -> None:
+        retry_delay = self.retry_backoff_seconds
+        while self.active:
+            try:
+                request = sync_pb2.SyncFlagsRequest(selector=self.selector)
+                logger.debug("Setting up gRPC sync flags connection")
+                for flag_rsp in self.stub.SyncFlags(request):
+                    flag_str = flag_rsp.flag_configuration
+                    logger.debug(
+                        f"Received flag configuration - {abs(hash(flag_str)) % (10 ** 8)}"
+                    )
+                    self.flag_store.update(json.loads(flag_str))
+
+                    if not self.connected:
+                        self.emit_provider_ready(
+                            ProviderEventDetails(
+                                message="gRPC sync connection established"
+                            )
+                        )
+                        self.connected = True
+                        # reset retry delay after successsful read
+                        retry_delay = self.retry_backoff_seconds
+                    if not self.active:
+                        logger.info("Terminating gRPC sync thread")
+                        return
+            except grpc.RpcError as e:
+                logger.error(f"SyncFlags stream error, {e.code()=} {e.details()=}")
+                if e.code() == grpc.StatusCode.UNAVAILABLE:
```

Comment:
destroying the existing channel, and creating a new one, actually fixed the reconnect issue. i am not sure, if this is best practice, or should be avoided.

Reply:
I wonder if we need to handle more cases than this? `UNKNOWN` perhaps? Just wondering if other sorts of network errors that result in the same bad state might not be handled because they return a different status.

Reply:
For me it is really hard to judge, my grpc knowledge is Limited, and I would fully rely on your opinion. If you do think it make sense let's go for it.

Reply:
My gut says since we don't 100% understand why we observe this behavior, we should play it safe and recreate the stub on any `grpc.RpcError` error; in particular I'm worried that if we don't, the stream deadline functionality might not work reliably.

I think there's not much risk here since this is only for the stream call and if we're having a problem with that, pretty much nothing else in this resolver mode is working anyway.

Reply:
Done here: https://github.com/open-feature/python-sdk-contrib/pull/104/commits/f91dd5c53310e646b0e7cec4a3c1c2a2a7f76ac0